lib/msf/core/payload/windows/reverse_http.rb

```diff
@@ -123,13 +136,37 @@ def asm_reverse_http(opts={})
 
     #
     # options should contain:
-    #    ssl:     (true|false)
-    #    url:     "/url_to_request"
-    #   host:     [hostname]
-    #   port:     [port]
-    # exitfunk:   [process|thread|seh|sleep]
+    #        ssl: (true|false)
+    #        url: "/url_to_request"
+    #       host: [hostname]
+    #       port: [port]
+    #   exitfunk: [process|thread|seh|sleep]
+    # proxy_host: [proxy-server]
+    # proxy_port: [port]
+    # proxy_user: [username]
+    # proxy_pass: [password]
+    # proxy_type: [HTTP|SOCKS]
     #
 
+    proxy_enabled = !!(opts[:proxy_host].to_s.strip.length > 0)
+    proxy_info    = ""
+
+    if proxy_enabled
+      if opts[:proxy_type].to_s.downcase == "socks"
+        proxy_info << "socks="
+      else
+        proxy_info << "http://"
+      end
+
+      proxy_info << opts[:proxy_host].to_s
+      if opts[:proxy_port].to_i > 0
+        proxy_info << ":#{opts[:proxy_port]}"
+      end
+    end
+
+    proxy_user = opts[:proxy_user].to_s.length == 0 ? nil : opts[:proxy_user]
```

Comment:
```NoMethodError: undefined method `blank?' for "":String``` (yeah yeah, ActiveSupport). An empty string might be a usable username or password though (corner case if so).